hudi-common/src/main/java/org/apache/hudi/avro/AvroSchemaUtils.java

```diff
@@ -206,6 +207,18 @@ private static boolean isProjectionOfInternal(Schema sourceSchema,
     return atomicTypeEqualityPredicate.apply(sourceSchema, targetSchema);
   }
 
+  public static Option<Schema.Type> findNestedFieldType(Schema schema, String fieldName) {
```

Comment:
Don't we already have a util to find nested field type? Can reuse some parts of `HoodieAvroUtils.getNestedFieldSchemaFromRecord`. In case a new method is needed, please also add a UT for the same.

Reply:
`HoodieAvroUtils .getNestedFieldSchemaFromRecord` uses the actual generic record. I would have thought `HoodieAvroUtils .getNestedFieldSchemaFromWriteSchema` would do exactly that, but it doesn't. I honestly confused af about what `TestHoodieAvroUtils.testGetNestedFieldSchema` is trying to do. It puts firstname and lastname as both top level and nested fields, and then it's just getting the schema from the top level fields and not the nested ones.